### PR TITLE
add doc build steps to contrib guide

### PR DESCRIPTION
**Pull request by kmaehashi**
Write steps to build docs for contributors.
(context: https://github.com/chainer/chainer/pull/4397#issuecomment-387044030)

### CODE REVIEW

**Comment by niboshi** [comment]
How about `pip install -e .[docs]`?

File: docs/source/contribution.rst [hunk 1]
```diff
@@ -398,3 +398,29 @@ Note that reviewers will test your code without the option to check CUDA-related
...
+
+  $ pip install sphinx sphinx_rtd_theme
+
+Then you can build the HTML document locally::
+
+  $ cd docs
+  $ make html
+
+HTML files are generated under ``build/html`` directory.
+Open it with the browser and see if it is rendered as expected.
```
Comment (niboshi) [inline docs/source/contribution.rst]: Maybe better to mention `index.html`?

**Comment by kmaehashi** [comment]
Thanks, fixed.

> How about pip install -e .[docs]?

Currently other part in this doc does not use `.[xxx]`:

> pip install autopep8 hacking
> pip install pytest mock

I think it's OK to keep as it is, because it is more clear to users, and it rarely changes.

File: docs/source/contribution.rst [hunk 2]
```diff
@@ -398,3 +398,34 @@ Note that reviewers will test your code without the option to check CUDA-related
...
+
+When adding a new feature to the framework, you also need to document it to the reference.
+For example, if you are adding a new function under ``chainer.functions``, you need to add it to the :doc:`reference/functions` page.
+
+.. note::
+
+   If you are unsure about how to fix the documentation, you can submit a pull request without doing so.
+   Reviewers will help you fix the documentation appropriately.
+
+The documentation is stored under `docs directory <https://github.com/chainer/chainer/tree/master/docs>`_ and written in `reStructuredText <http://www.sphinx-doc.org/en/master/usage/restructuredtext/index.html>`_ format.
```
Comment (niboshi) [inline docs/source/contribution.rst]: "The documentation source"?

File: docs/source/contribution.rst [hunk 3]
```diff
@@ -398,3 +398,34 @@ Note that reviewers will test your code without the option to check CUDA-related
...
+
+To build the documentation, you need to install `Sphinx <http://www.sphinx-doc.org/>`_::
+
+  $ pip install sphinx sphinx_rtd_theme
+
+Then you can build the HTML document locally::
```
Comment (niboshi) [inline docs/source/contribution.rst]: "the documentation in HTML format"?

File: docs/source/contribution.rst [hunk 4]
```diff
@@ -398,3 +398,34 @@ Note that reviewers will test your code without the option to check CUDA-related
 .. note::
    Some of numerically unstable tests might cause errors irrelevant to your changes.
    In such a case, we ignore the failures and go on to the review process, so do not worry about it!
+
+
+Documentation
+-------------
+
+When adding a new feature to the framework, you also need to document it to the reference.
```
Comment (niboshi) [inline docs/source/contribution.rst]: to the reference -> in the reference?

**Comment by kmaehashi** [comment]
Fixed accordingly.

**Comment by niboshi** [comment]
LGTM!

**Pull request merged**